src/librustdoc/clean/mod.rs

```diff
@@ -57,11 +57,43 @@ impl<'tcx> Clean<'tcx, Item> for DocModule<'tcx> {
                 .map(|(item, renamed)| clean_maybe_renamed_foreign_item(cx, item, *renamed)),
         );
         items.extend(self.mods.iter().map(|x| x.clean(cx)));
-        items.extend(
-            self.items
-                .iter()
-                .flat_map(|(item, renamed)| clean_maybe_renamed_item(cx, item, *renamed)),
-        );
+
+        // Split up imports from all other items.
+        //
+        // This covers the case where somebody does an import which should pull in an item,
+        // but there's already an item with the same namespace and same name. Rust gives
+        // priority to the not-imported one, so we should, too.
```

Comment:
I'd suggest mentioning this is specifically about glob imports. For normal imports, the resolver gives an error that you have duplicate items: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=130848897ca4a33ebafad5fa75ae8884
```
error[E0255]: the name `S` is defined multiple times
```